org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/rule/RuleDetailsPanel.java

```diff
@@ -111,32 +117,32 @@ private void updateScrollCompositeMinSize() {
   }
 
   public void updateRule(GetStandaloneRuleDescriptionResponse getStandaloneRuleDescriptionResponse) {
-    try {
-      var ruleDefinition = getStandaloneRuleDescriptionResponse.getRuleDefinition();
-
-      ruleNameLabel.setText(ruleDefinition.getName());
-      ruleNameLabel.requestLayout();
-      ruleHeaderPanel.updateRule(ruleDefinition.getKey(), ruleDefinition.getType(), ruleDefinition.getDefaultSeverity());
-
-      updateHtmlDescription(getStandaloneRuleDescriptionResponse.getDescription(), ruleDefinition.getLanguage().getLanguageKey());
-
-      requestLayout();
-      updateScrollCompositeMinSize();
-    } catch (SWTException ignored) {
-      // There might be a race condition between the background job running late and the view already being closed
-    }
+    var orig = getStandaloneRuleDescriptionResponse.getRuleDefinition();
+    var fake = new RuleDefinitionDto(orig.getKey(), orig.getName(), orig.getSeverity(), orig.getType(),
+      CleanCodeAttribute.COMPLETE, Map.of(SoftwareQuality.SECURITY, ImpactSeverity.HIGH, SoftwareQuality.MAINTAINABILITY, ImpactSeverity.MEDIUM, SoftwareQuality.RELIABILITY, ImpactSeverity.LOW),
+      orig.getParamsByKey(), orig.isActiveByDefault(), orig.getLanguage());
+    
+    updateRule(fake,
+      getStandaloneRuleDescriptionResponse.getDescription());
   }
 
   public void updateRule(GetEffectiveRuleDetailsResponse getEffectiveRuleDetailsResponse) {
+    var details = getEffectiveRuleDetailsResponse.details();
+    updateRule(details, details.getDescription());
+  }
+  
+  private void updateRule(AbstractRuleDto ruleInformation, Either<RuleMonolithicDescriptionDto, RuleSplitDescriptionDto> description) {
     try {
-      var details = getEffectiveRuleDetailsResponse.details();
-
-      ruleNameLabel.setText(details.getName());
+      ruleNameLabel.setText(ruleInformation.getName());
       ruleNameLabel.requestLayout();
-      ruleHeaderPanel.updateRule(details.getKey(), details.getType(), details.getSeverity());
+      
+      ruleHeaderPanel.updateRule(ruleInformation);
 
-      updateHtmlDescription(details.getDescription(), details.getLanguage().getLanguageKey());
-      updateParameters(details);
+      updateHtmlDescription(description, ruleInformation.getLanguage().getLanguageKey());
+      
+      if (ruleInformation instanceof EffectiveRuleDetailsDto) {
+        updateParameters((EffectiveRuleDetailsDto) ruleInformation);
+      }
```

Comment:
Sadly no because we have to run `requestLayout()` and `updateScrollCompositeMinSize()` afterward, therefore moving it to the other method would move this out of the shared method as well. Additionally, the try-catch block also applies to this `updateParameters(...)` call and so it would have to be moved to the separate methods as well.